src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -55,6 +54,78 @@ public static Task StopAsync(this IWebHost host, TimeSpan timeout)
         }
 
         /// <summary>
+        /// Block the calling thread until shutdown is triggered via Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        public static void WaitForShutdown(this IWebHost host)
+        {
+            host.WaitForShutdownAsync().GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Block the calling thread until shutdown is triggered via the given token, Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        /// <param name="token">The token to trigger shutdown.</param>
+        public static void WaitForShutdown(this IWebHost host, CancellationToken token)
+        {
+            host.WaitForShutdownAsync(token).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Returns a Task that completes when shutdown is triggered via Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        public static Task WaitForShutdownAsync(this IWebHost host)
+        {
+            return host.WaitForShutdownAsync(CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Returns a Task that completes when shutdown is triggered via the given token, Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IWebHost"/>.</param>
+        /// <param name="token">The token to trigger shutdown.</param>
+        public static async Task WaitForShutdownAsync(this IWebHost host, CancellationToken token)
```

Comment:
Not sure if that's worth the effort. There are a few differences:

1. RunAsync(CancellationToken) triggers shutdown only based on the token whereas WaitForShutdownAsync reacts to CTRL+C and SIGTERM by default. I'm not sure if this is a behaviour change we want.
2. RunAsync() matches the behaviour of WaitForShutdownAsync more but it calls RunAsync(CancellationToken) currently and takes advantage of the console logging there. If we were to use WaitForShutdownAsync instead, we need to duplicate a lot of the logging. 

For these reasons, I think it might be better to keep RunAsync and WaitForShutdown separate.